src/main/java/de/tum/in/www1/artemis/service/connectors/BitbucketBambooUpdateService.java

```diff
@@ -0,0 +1,112 @@
+                bambooRemoteRepository = bambooClient.getRepositoryHelper().getRemoteRepository("Assignment", planKey, false);
+                if (bambooRemoteRepository == null) {
+                    throw new BambooException("Something went wrong while updating the template repository of the build plan " + planKey
+                            + " to the student repository : Could not find assignment nor Assignment repository");
+                }
+            }
+
+            bambooBuildPlanUpdateProvider.updateRepository(bambooRemoteRepository, bitbucketRepository, bitbucketProject, planKey);
+
+            // Overwrite triggers if needed, incl workaround for different repo names
+            if (triggeredBy.isPresent() && bambooRemoteRepository.getName().equals("Assignment")) {
+                triggeredBy = Optional.of(triggeredBy.get().stream().map(trigger -> trigger.replace(Constants.ASSIGNMENT_REPO_NAME, "Assignment")).collect(Collectors.toList()));
+            }
+            triggeredBy.ifPresent(repoTriggers -> overwriteTriggers(planKey, bambooClient, repoTriggers));
+
+            log.info("Update plan repository for build plan " + planKey + " was successful");
+        }
+        catch (CliClient.ClientException | CliClient.RemoteRestException e) {
+            throw new BambooException("Something went wrong while updating the template repository of the build plan " + planKey + " to the student repository : " + e.getMessage(),
+                    e);
+        }
+    }
+
+    private void overwriteTriggers(final String planKey, final BambooClient bambooClient, final List<String> triggeredBy) {
+        try {
+            final var triggersString = bambooClient.getTriggerHelper().getTriggerList(planKey, null, null, 99, Pattern.compile(".*"));
+            // Bamboo CLI returns a weird String, which is the reason for this way of parsing it
+            final var oldTriggers = Arrays.stream(triggersString.split("\n")).map(trigger -> trigger.replace("\"", "").split(","))
+                    .filter(trigger -> trigger.length > 2 && NumberUtils.isCreatable(trigger[1])).map(trigger -> Long.parseLong(trigger[1])).collect(Collectors.toSet());
+
+            // Remove all old triggers
+            for (final var triggerId : oldTriggers) {
+                bambooClient.getTriggerHelper().removeTrigger(planKey, null, null, triggerId, null, false);
+            }
+
+            // Add new triggers
+            for (final var repo : triggeredBy) {
+                bambooClient.getTriggerHelper().addTrigger(planKey, null, "remoteBitbucketServer", null, null, repo, null, null, false);
+            }
+        }
+        catch (CliClient.ClientException | CliClient.RemoteRestException e) {
+            throw new BambooException("Unable to overwrite triggers for " + planKey + "\n" + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void triggerUpdate(String buildPlanId, boolean initialBuild) {
+        // NOT NEEDED
```

Comment:
Why not needed?

Reply:
I didn't write this code and I also don't really get why this is here :smile: . The JavaDoc also doesn't really match the by the method name implied usage. I would suggest to maybe remove this in a future PR

Reply:
I guess this is outdated. The corresponding method `triggerUpdate` in `ContinuousIntegrationUpdateService` is not used. I guess the trigger build was implemented differently. Please remove it.

Reply:
I just removed the outdated code